xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-notifiers/xwiki-platform-notifications-notifiers-default/src/main/java/org/xwiki/notifications/notifiers/internal/UserEventDispatcher.java

```diff
@@ -280,15 +281,15 @@ public void dispatch(Event event)
         }
     }
 
-    private void dispatchInContext(Event event)
+    private void dispatchInContext(Event event) throws UserException
```

Comment:
I'm wondering if it's valid to fail the entire pre-filtering if one of the user profile page is broken.

Problem is that it all depend what is the cause:
* general problem: should definitely stop (i.e. not mark the event as pre-filtered)
* only one of the users have a broken document: quite a pain to penalize all following users

Reply:
I agree. My assumption is that the exception will only occur for very bad failures where all users are impacted. Also, if we just ignore "failed" users, we have no easy way to allow user the re-execute the migration.